File: native/strapiclient/src/model/article.rs
use crate::model::GalleryImage;
use rustler::NifStruct;

#[derive(Debug, NifStruct)]
#[module = "Kotkowo.Article"]
pub struct Article {
    pub title: String,
    pub introduction: String,
    pub content: String,
    pub image: GalleryImage,
}

File: lib/kotkowo/article.ex
defmodule Kotkowo.Article do
  @moduledoc false

  @type t :: %__MODULE__{
          introduction: String.t(),
          image: GalleryImage,
          content: String.t()
        }
  defstruct [
    :introduction,
    :image,
    :content
  ]
end
